sound/soc/sof/loader.c

```diff
@@ -388,6 +388,29 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 		return ret;
 	}
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_IPC_FLOOD_TEST)
+	if (sdev->first_boot) {
+		/* create read-write ipc_flood_count debugfs entry */
+		ret = snd_sof_debugfs_buf_item(sdev, NULL, 0,
+					       "ipc_flood_count", 0666);
+
+		/* errors are only due to memory allocation, not debugfs */
+		if (ret < 0) {
+			dev_err(sdev->dev, "error: snd_sof_debugfs_buf_item failed\n");
+			return ret;
+		}
+
+		/* create read-write ipc_flood_duration_ms debugfs entry */
+		ret = snd_sof_debugfs_buf_item(sdev, NULL, 0,
+					       "ipc_flood_duration_ms", 0666);
+
+		/* errors are only due to memory allocation, not debugfs */
+		if (ret < 0) {
+			dev_err(sdev->dev, "error: snd_sof_debugfs_buf_item failed\n");
+			return ret;
+		}
+	}
+#endif
```

Comment:
nit-pick: do we need this code here? Couldn't it be moved for debug.c for consistency? This really has nothing to do with the loader, does it?

Reply:
or could it be moved earlier in the function to have all the debugfs stuff in one place. Here it's in the middle of unrelated code, it's a bit odd.

Reply:
@plbossart makes sense. I've moved it to the debugfs init function. I think the fw_version creation can also be moved there. But I'll leave it in as is for now